corehq/apps/userreports/management/commands/rename_ucr_tables.py

```diff
@@ -0,0 +1,109 @@
+from django.core.management.base import BaseCommand
+from six.moves import input
+
+from corehq.apps.userreports.models import DataSourceConfiguration, StaticDataSourceConfiguration
+from corehq.apps.userreports.util import get_table_name
+from corehq.sql_db.connections import connection_manager
+from dimagi.utils.couch.database import iter_docs
+
+OLD_UCR_TABLE_PREFIX = 'config_report_'
+
+
+def _table_names(domain, table_id):
+    old_table = get_table_name(domain, table_id, max_length=63, prefix=OLD_UCR_TABLE_PREFIX)
+    new_table = get_table_name(domain, table_id)
+    return old_table, new_table
+
+
+def _get_old_new_tablenames():
+    data_source_ids = [
+        row['id']
+        for row in DataSourceConfiguration.view(
+            'userreports/active_data_sources', reduce=False, include_docs=False
+        )
+    ]
+    by_engine_id = defaultdict(list)
+    for ds in iter_docs(DataSourceConfiguration.get_db(), data_source_ids):
+        by_engine_id[ds['engine_id']].append(_table_names(ds['domain'], ds['table_id']))
+
+    for ds in StaticDataSourceConfiguration.all():
+        by_engine_id[ds['engine_id']].append(_table_names(ds['domain'], ds['table_id']))
+
+    return by_engine_id
+
+
+def _table_exists(conn, table_name):
+    res = conn.execute("select 1 from pg_tables where tablename = %s", table_name)
+    return bool(list(res))
+
+
+def _add_view(conn, table_name, view_name):
+    if _table_exists(conn, table_name):
+        res = conn.execute("select 1 from pg_views where viewname = %s", view_name)
+        view_exists = bool(list(res))
+        return not view_exists
+    return False
+
+
+def create_ucr_views():
+    tables_by_engine = _get_old_new_tablenames()
+    for engine_id, tables in tables_by_engine.items():
+        print('\tChecking {} tables in engine "{}"'.format(len(tables), engine_id))
+        engine = connection_manager.get_engine(engine_id)
+        view_creates = []
+        with engine.begin() as conn:
+            for old_table, new_table in tables:
+                if _add_view(conn, old_table, new_table):
+                    view_creates.append('CREATE VIEW "{}" AS SELECT * FROM "{}";'.format(new_table, old_table))
+
+        if view_creates:
+            print('\tCreating {} views in engine "{}"'.format(len(view_creates), engine_id))
+            with engine.begin() as conn:
+                conn.execute('\n'.join(view_creates))
+        else:
+            print('\tNo views to create in engine "{}"'.format(engine_id))
+
+
+def _rename_tables():
+    tables_by_engine = _get_old_new_tablenames()
+    for engine_id, tables in tables_by_engine.items():
+        engine = connection_manager.get_engine(engine_id)
+        print('\tChecking {} tables in engine "{}"'.format(len(tables), engine_id))
+        for old_table, new_table in tables:
+            drop_view_rename_table = """
+            DROP VIEW IF EXISTS "{new_table}";
+            ALTER TABLE "{old_table}" RENAME TO "{new_table}";
```

Comment:
Not a big issue, but why not swap these two lines. If it were to drop and then fail for whatever reason, there would be a missing table for a while?

Reply:
You can't do that since we're renaming the table to the same name as the view. Since this is happening in a transaction it should appear atomic to other clients.